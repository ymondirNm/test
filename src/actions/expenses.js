import uuid from "uuid";
import database from "../firebase/firebase";

//ADD EXPENSE
export const addExpense = expense => ({
  type: "ADD_EXPENSE",
  expense
});

//REMOVE EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
});

//EDIT EXPENSE
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates
});

export const startAddExpense = (expanseData = []) => {
  return dispatch => {
    const {
      description = "",
      note = "",
      amount = 0,
      createdAt = 0
    } = expanseData;
    const expense = { description, note, amount, createdAt };
    return database
      .ref("expenses")
      .push(expense)
      .then(ref => {
        dispatch(
          addExpense({
            id: ref.key,
            ...expense
          })
        );
      });
  };
};

// SET_EXPENSES
export const setExpenses = expenses => ({
  type: "SET_EXPENSES",
  expenses
});
// export const startSetExpenses;
export const startSetExpenses = () => {
  return dispatch => {
    return database
      .ref("expenses")
      .once("value")
      .then(snapshot => {
        const expensesData = [];
        snapshot.forEach(child => {
          expensesData.push({
            id: child.key,
            ...child.val()
          });
        });

        dispatch(setExpenses(expensesData));
      });
  };
};

// export const startSetExpenses;
export const startRemoveExpense = id => {
  return dispatch => {
    return database
      .ref(`expenses/${id}`)
      .set(null)
      .then(() => {
        dispatch(removeExpense({ id }));
      });
  };
};
// export const startSetExpenses;
export const startEditExpense = (id, updates) => {
  return dispatch => {
    return database
      .ref(`expenses/${id}`)
      .update(updates)
      .then(() => {
        dispatch(editExpense(id, updates));
      });
  };
};
