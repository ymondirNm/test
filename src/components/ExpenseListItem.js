import React from "react";
import { Link } from "react-router-dom";
import moment from 'moment';
import numeral  from 'numeral';
// load a locale
numeral.register('locale', 'fr', {
    delimiters: {
        thousands: ' ',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '€'
    }
});

// switch between locales
numeral.locale('fr');
export const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>
        {numeral(amount).format('$0,0.00')}
        -
        {moment(createdAt).format('Do MMMM YYYY')}
      </p>
    </div>
  );
};
export default ExpenseListItem;
