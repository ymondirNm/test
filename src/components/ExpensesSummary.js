import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>
    <h1>
      Viewing {expenseCount} {expenseCount === 1 ? "expense" : "expenses"}{" "}
      totalling {numeral(expensesTotal/100).format("$0,0.00")}
    </h1>
  </div>
);

const mapStateToProps = state => {
    const selectedExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: selectedExpenses.length,
    expensesTotal: getExpensesTotal(selectedExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
