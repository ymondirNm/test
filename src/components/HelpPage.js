import React from 'react';

const HelpPage = () => (
    <div>this is from the HelpPage</div>
);

export default HelpPage