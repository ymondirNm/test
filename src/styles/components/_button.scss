.big-button {
  background: $blue;
  border: none;
  border-bottom: 0.6rem solid darken($blue, 10%);
  color: #fff;
  font-weight: bold;
  font-size: $l-size;
  width: 100%;
  margin-bottom: $m-size;
  padding: 2.4rem;

  &:disabled {
    opacity: 0.5;
  }
}
@media (min-width: $desktop-breakpoint) {
  .big-button {
    margin-bottom: $xl-size;
  }
}

.button {
  background: $blue;
  border: none;
  // border-bottom: .3rem solid darken($blue, 10%);
  color: #fff;
  font-weight: 300;
  padding: $s-size;
  font-size: $font-size-large;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none; 
}

.button--secondary {
  background: $gray; 
}
