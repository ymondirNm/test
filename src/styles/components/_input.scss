.baseInput {
  border: 1px solid #cacccd;
  height: 50px;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}

.text-input {

  @extend .baseInput;

}

.select {
  @extend .baseInput;
}

.textarea {
  @extend .baseInput;
  height: 10rem;
}


