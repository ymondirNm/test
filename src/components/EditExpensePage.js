import React, { Component } from 'react';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    onClick = () => {
        this.props.removeExpense(this.props.expense.id);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>

                <h1>Edit Expanse</h1>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />

                <button onClick={this.onClick}>Remove</button>
            </div>
        )
    }

}


const mapStateToProps = (state, props) => {
    const expense = state.expenses.find((expense) => expense.id === props.match.params.id);
    if (expense) {
        return { expense };
    } else {
        props.history.push('/notFound')
        return {}
    }
    /*   return {
           expense: state.expenses.find((expense) => expense.id === props.match.params.id)
       }*/
};

const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (id) => dispatch(removeExpense({ id }))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);