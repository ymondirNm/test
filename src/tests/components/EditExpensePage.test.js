import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';
import moment from 'moment';

let editExpense,removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage removeExpense={removeExpense} editExpense={editExpense} history={history} expense={expenses[1]}/>);
})

test('should render EditExpensePage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id,expenses[1]);
});

test('should handle removeExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith(expenses[1].id); 
});