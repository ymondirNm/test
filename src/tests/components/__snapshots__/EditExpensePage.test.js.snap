// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <h1>
    Edit Expanse
  </h1>
  <ExpenseForm
    expense={
      Object {
        "amount": 125000,
        "createdAt": -345600000,
        "description": "Rent bill",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
